pool/pool_router.gno

```diff
@@ -0,0 +1,124 @@
+package pool
+
+import (
+	"strconv"
+	"strings"
+
+	"gno.land/p/demo/ufmt"
+)
+
+func FindBestPool(
+	tokenA string,
+	tokenB string,
+	zeroForOne bool,
+	amountSpecified bigint,
+) string {
+	if tokenA == tokenB {
+		panic("token pair cannot be the same")
+	}
+
+	if tokenA > tokenB {
+		tokenA, tokenB = tokenB, tokenA
+	}
+
+	partialPath := tokenA + "_" + tokenB + "_"
+	foundPool := []string{}
+
+	for poolPath, _ := range pools {
+		if strings.HasPrefix(poolPath, partialPath) {
+			foundPool = append(foundPool, poolPath)
+		}
+	}
+
+	// check if amount is enough
+	firstSelectedPool := []string{}
+	if false {
+		continue
+	} else if zeroForOne == true && amountSpecified > 0 {
```

Comment:
is there no case amountSpecified is zero?

Reply:
yes, will panic if amountSpecified is `0`